sphinx/domains/javascript.py

```diff
@@ -37,10 +37,10 @@ class JSObject(ObjectDescription[Tuple[str, str]]):
     #: based on directive nesting
     allow_nesting = False
 
-    option_spec: OptionSpec = {
-        'noindex': directives.flag,
+    option_spec: OptionSpec = ObjectDescription.option_spec.copy()
+    option_spec.update({
```

Comment:
Please just copy `:hiden:` to all subclasses, it makes it easier to reason about.

Reply:
> Please just copy :hiden: to all subclasses, it makes it easier to reason about.

Here I followed the examples given py `sphinx/domains/python.py` et al as well as the
docutils classes, which (except for the Tables) copy the `option_spec` of their
super class end enhance it.  This IMHO makes it sense since – like for example
in this PR – we can add an option/functionality to all object description
directives simply by adding it to the super class `ObjectDescription` without
modifying the (potentially many) subclasses.  Isn't that what inheritance is
about?

If you still think it makes sense to verbatim copy the `option_spec` values,
please say so and I'll change it.  In that case: Should I also update all the
current `option_spec.copy()` not touched by this PR, i.e. in
`sphinx/domains/rst.py` and `sphinx/domains/python.py`?

Reply:
Makes sense, the current method (`.update`) is fine.

A